docs/profiles.md

```diff
@@ -13,12 +27,15 @@ For argument's sake, we'll want these changes in CI:
 - The URL for the app (maybe dynamically provisioned?)
 - The formatters we want to use
 
-To start using Profiles, we just need to create a `cucumber.js` file in our project's root. It's a simple JavaScript module that exports an object with profile names as keys and CLI options as values. We can lean on JavaScript to reduce duplication and grab things dynamically as needed. Here's what we might write to address the needs described above:
+To start using Profiles, we just need to create a `cucumber.js` file in our project's root. This file must be a CommonJS module, so when using Cucumber with ES6 modules the file will need to be named `cucumber.cjs`. 
```

Comment:
I think `.cjs` extension works everytime. What do you think to use it all the time and not mentioning the `.js` one at all?

Reply:
You're right, but it might be too subtle. Consider someone just flat out missing the c and entering in 'cucumber.js' out of habit, especially if upgrading. I can see a dev spending time and getting frustrated up until they noticed their mistake.  With the wording above the reason for the problem is made more immediately clear.

Reply:
Maybe a "UPGRADING.md" document is missing to bring the info to experienced users?

Reply:
I'll just note that the config file extension is also called out in the dedicated page for ESM https://github.com/cucumber/cucumber-js/blob/main/docs/esm.md, so I think between that and this we're covering the bases well enough.

Also as part of https://github.com/cucumber/cucumber-js/issues/751 I think it's likely we'll add a new config file format without the compatibility issues (which come from having `default` as one of the keys). But this is a good documentation of the current state of things.

Reply:
That still wouldn't alleviate the subtlety issue. Besides, an Upgrading doc would be a moving target.  I'm ok with telling the user to use a cjs file at all times though